backend/infrahub/core/manager.py

```diff
@@ -569,41 +715,64 @@ async def get_one_by_hfid(
         db: InfrahubDatabase,
         hfid: list[str],
         kind: str,
-        fields: Optional[dict] = None,
-        at: Optional[Union[Timestamp, str]] = None,
-        branch: Optional[Union[Branch, str]] = None,
-        include_source: bool = False,
-        include_owner: bool = False,
-        prefetch_relationships: bool = False,
-        account=None,
-        branch_agnostic: bool = False,
-        raise_on_error: Literal[True] = True,
+        raise_on_error: Literal[True],
+        fields: dict | None = ...,
+        at: Timestamp | str | None = ...,
+        branch: Branch | str | None = ...,
+        include_source: bool = ...,
+        include_owner: bool = ...,
+        prefetch_relationships: bool = ...,
+        account=...,
+        branch_agnostic: bool = ...,
     ) -> Any: ...
 
+    @overload
     @classmethod
     async def get_one_by_hfid(
         cls,
         db: InfrahubDatabase,
         hfid: list[str],
         kind: str,
-        fields: Optional[dict] = None,
-        at: Optional[Union[Timestamp, str]] = None,
-        branch: Optional[Union[Branch, str]] = None,
+        raise_on_error: bool = ...,
+        fields: dict | None = ...,
+        at: Timestamp | str | None = ...,
+        branch: Branch | str | None = ...,
+        include_source: bool = ...,
+        include_owner: bool = ...,
+        prefetch_relationships: bool = ...,
+        account=...,
+        branch_agnostic: bool = ...,
+    ) -> Any: ...
+
+    @classmethod
+    async def get_one_by_hfid(
+        cls,
+        db: InfrahubDatabase,
+        hfid: list[str],
+        kind: type[SchemaProtocol] | str,
+        raise_on_error: bool = False,
+        fields: dict | None = None,
+        at: Timestamp | str | None = None,
+        branch: Branch | str | None = None,
         include_source: bool = False,
         include_owner: bool = False,
         prefetch_relationships: bool = False,
         account=None,
         branch_agnostic: bool = False,
-        raise_on_error: bool = False,
-    ) -> Optional[Any]:
+    ) -> Any:
         branch = await registry.get_branch(branch=branch, db=db)
         at = Timestamp(at)
 
+        if hasattr(kind, "_is_runtime_protocol") and getattr(kind, "_is_runtime_protocol"):
+            kind_str = kind.__name__
+        else:
+            kind_str = kind
```

Comment:
Perhaps we should have a function for this too..

Reply:
Done